examples/pytorch/audio-classification/run_audio_classification.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+import logging
+import os
+import sys
+from dataclasses import dataclass, field
+from random import randint
+from typing import Optional
+
+import datasets
+import numpy as np
+import torchaudio
+from datasets import load_dataset
+
+import transformers
+from transformers import (
+    MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING,
+    AutoConfig,
+    AutoModelForAudioClassification,
+    HfArgumentParser,
+    Trainer,
+    TrainingArguments,
+    Wav2Vec2FeatureExtractor,
+)
+from transformers.trainer_utils import get_last_checkpoint
+from transformers.utils import check_min_version
+from transformers.utils.versions import require_version
+
+
+logger = logging.getLogger(__name__)
+
+# Will error if the minimal version of Transformers is not installed. Remove at your own risks.
+check_min_version("4.11.0.dev0")
+
+require_version("datasets>=1.12.1", "To fix: pip install -r examples/pytorch/audio-classification/requirements.txt")
+
+MODEL_CONFIG_CLASSES = list(MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING.keys())
+MODEL_TYPES = tuple(conf.model_type for conf in MODEL_CONFIG_CLASSES)
+
+
+def load_audio(path: str, sample_rate: int = 16000):
+    wav, sr = torchaudio.load(path)
+    # convert multi-channel audio to mono
+    wav = wav.mean(0)
+    # standardize sample rate if it varies in the dataset
+    resampler = torchaudio.transforms.Resample(sr, sample_rate)
+    wav = resampler(wav)
+    return wav
+
+
+def random_subsample(wav: np.ndarray, max_length: float, sample_rate: int = 16000):
```

Comment:
Does this improve performance you think? Randomly subsampling instead of simple cutting with truncate?

Reply:
As discussed, in Keyword Spotting there is a `_silence_` class that uses 10min audio clips to sample from, and random sampling is suggested by the SUPERB authors there.